defs.bzl

```diff
@@ -74,6 +74,19 @@ def _convert_js_files_to_amd_modules(ctx, js_protoc_outputs):
 
     return js_outputs
 
+def get_output_dir(ctx, outputs):
+    # Default to using bin_dir
+    output_dir = ctx.bin_dir.path
+
+    if len(outputs) <= 0:
+        return output_dir
+
+    declared_output = outputs[0]
+    if declared_output.dirname.endswith(ctx.label.package):
```

Comment:
Could you add a comment here to describe what this check is doing, maybe by documenting which path formats are expected to be seen here?